### PR TITLE
[cxx-interop][android] link with CxxStdlib when building for Android too

### PR DESCRIPTION
**Pull request by hyp**

### CODE REVIEW

**Comment by hyp** [comment]
Still needs a test, but testing on adopters now.

**Comment by hyp** [comment]
@swift-ci please smoke test

**Comment by hyp** [comment]
CC @finagolfin

File: lib/AST/ModuleDependencies.cpp [hunk 1]
```diff
@@ -536,19 +536,14 @@ swift::dependencies::registerCxxInteropLibraries(
...
-      break;
     case llvm::Triple::Win32: {
       RegistrationCallback(
           LinkLibrary(hasStaticCxxStdlib ? "libswiftCxxStdlib" : "swiftCxxStdlib",
                       LibraryKind::Library));
       break;
     }
     default:
-      if (Target.isOSDarwin())
+      if (Target.isOSDarwin() || Target.isOSLinux())
```
Comment (compnerd) [inline lib/AST/ModuleDependencies.cpp:546]: I guess that BSD is still out otherwise it would be nice to collapse this switch.

**Review by compnerd — APPROVED** [review]

**Comment by finagolfin** [comment]
I don't know much about this, as I don't use C++ Interop myself, and never noticed this affecting the Interop tests in the compiler validation suite. Now that you enabled building this library for Android in an earlier pull, I had a problem building this library in Termux with the latest June 13 trunk snapshot because of a C++ Interop error, so I disabled building it again for now, but will try it again soon. If you need it and it works for you, this pull sounds good to me.

Speaking of the latest trunk source snapshot, I was finally able to build it and run the compiler validation suite and I'm happy to report that your Bionic modularization fixed a bunch of C++ Interop tests natively in Termux also, once I elided [a few Bionic header changes in the Termux app](https://github.com/termux/termux-packages/blob/master/ndk-patches/26b/stdio.h.patch#L7), including the `stdint.h` issue I mentioned earlier this week. I will clean everything up next and submit a pull enabling several C++ Interop tests that were previously disabled for Android, since they pass now. 😃

Any reason you haven't submitted your Android stdlib changes for the 6.0 branch yet?

**Review by hjyamauchi — APPROVED** [review]

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
It looks like some of the existing C++ interop runtime tests will cover this for Android.

**Comment by hyp** [comment]
@swift-ci please test macOS platform

**Pull request merged**